### PR TITLE
KAFKA-13101: Replace EasyMock and PowerMock with Mockito for RestServerTest

### PR DESCRIPTION
**Pull request by tang7526**
https://issues.apache.org/jira/browse/KAFKA-13101

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Thanks for the PR, we should also update https://github.com/apache/kafka/blob/trunk/build.gradle#L373

**Comment by tang7526** [comment]
> Thanks for the PR, we should also update https://github.com/apache/kafka/blob/trunk/build.gradle#L373

@ijuma Copy that. Done.

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java [hunk 1]
```diff
@@ -61,18 +56,23 @@
...
+    
     private Herder herder;
-    @MockStrict
     private Plugins plugins;
     private RestServer server;
 
     protected static final String KAFKA_CLUSTER_ID = "Xbafgnagvar";
 
+    @Before
+    public void tearUp() {
```
Comment (ijuma) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java]: This method is usually called `setUp()`.

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM, just one nit. I'll fix that and then merge it.

**Comment by ijuma** [comment]
Tests passed for JDK 8, 11 and 17. Merging to master.

**Comment by ijuma** [comment]
cc @rhauch @kkonstantine

**Pull request merged**